### ISSUE TITLE
Efficiently track pixels in SSE blitters

### ISSUE THREAD

**Issue by Starbuck5** [pygame/pygame#3358]
A couple weeks ago, I found a way to increase the performance of the AVX blitters: https://github.com/pygame/pygame/pull/3294

TLDR: This issue is to do that to the SSE blitters.

But I'm writing this issue for someone who probably doesn't know any of the context of what this is.
Pygame is written on top of SDL, but our own implementation of `blit` for surfaces with alpha, to maintain compatibility with SDL1, and to support some additional things, like blend modes.
To make these faster, we use SIMD -- https://en.wikipedia.org/wiki/Single_instruction,_multiple_data. Specifically, we have a layer of blit routines that support the SSE2 instruction set, and another set that support the newer and more powerful AVX2 instruction set.

To accomplish this:
- [ ] copy my optimization into simd_blitters_sse2.c
- [ ] figure out a way to run the SSE2 blitters explicitly (your system almost definitely supports AVX2, so it will try to use those instead) (my suggestion is to #define __AVX2__ 0 in simd_blitters_avx2.c)
- [ ] benchmark performance and see how this optimization does on them

Note: This bit of optimizable code might not be present in every blitter, as I look through them... I did see one at least, there are definitely more: https://github.com/pygame/pygame/blob/main/src_c/simd_blitters_sse2.c#L1152-L1155

@PurityLake has dibs on this issue, if they choose to handle it.

**Comment by PurityLake** [comment]
I'm up to the challenge

**Comment by MyreMylar** [comment]
Let me know if you have any issues.

The more people vaguely familiar with the blitting code the better. There is always more work in the blitting mines.
